### PR TITLE
Join merge attempt

### PR DESCRIPTION
**Pull request by EricLarson2020**
Type of change made:
 New feature (non-breaking change which adds functionality)
Added user product many too many relationship and joins table. Added kept to userProducts joins table. Set a one to many relationship with user and UserProduct.

 Testing - (enter which file here)

Detailed Description
Why is this change required? What problem does it solve?
Were there any challenges that arose while implementing this feature? If so, how were the addressed?
Files modified:
 all files here
code/api/src/modules/user/model.js
code/api/src/modules/user/resolvers.js
code/api/src/modules/user_product/model.js
code/api/src/modules/user_product/types.js
code/api/src/setup/schema/queries.js

### CODE REVIEW

File: code/api/src/migrations/7-user_products.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
+module.exports = {
```
Comment (EricLarson2020) [inline code/api/src/migrations/7-user_products.js:1]: This is the migration for UserProducts joins table, has kept which determines if products is kept

File: code/api/src/modules/product/model.js [hunk 2]
```diff
@@ -2,7 +2,7 @@
 
 // Product
 module.exports = function(sequelize, DataTypes) {
-  return sequelize.define('products', {
+  let Product = sequelize.define('products', {
```
Comment (EricLarson2020) [inline code/api/src/modules/product/model.js:5]: Changes to add belongs to many relationship to products connecting it to joins table

File: code/api/src/modules/user/model.js [hunk 3]
```diff
@@ -19,7 +19,9 @@ module.exports = function(sequelize, DataTypes) {
 
   User.associate = function(models) {
     User.hasMany(models.Subscription)
```
Comment (EricLarson2020) [inline code/api/src/modules/user/model.js:21]: joins relationship, belongs to many requires the through to connect it to joins table
the as in the has many connect the userProducts to user in the resolver

File: code/api/src/modules/user/resolvers.js [hunk 4]
```diff
@@ -103,7 +103,13 @@ export async function updateImageResolver(parentValue, { id, address }, { auth }
 
 // Get by ID
 export async function getById(parentValue, { id }) {
-  return await models.User.findOne({ where: { id } })
```
Comment (EricLarson2020) [inline code/api/src/modules/user/resolvers.js:106]: This allows your to query products and UserProduct on user

File: code/api/src/modules/user/types.js [hunk 5]
```diff
@@ -1,42 +1,42 @@
 // Imports
-import { GraphQLObjectType, GraphQLString, GraphQLInt } from 'graphql'
```
Comment (EricLarson2020) [inline code/api/src/modules/user/types.js:2]: data types to allow for access of various types of data

File: code/api/src/modules/user_product/model.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+'use strict'
```
Comment (EricLarson2020) [inline code/api/src/modules/user_product/model.js:1]: joins table model with corresponding relations

File: code/api/src/setup/schema/queries.js [hunk 7]
```diff
@@ -17,7 +18,8 @@ const query = new GraphQLObjectType({
     ...product,
     ...crate,
     ...subscription
+
   })
 })
```
Comment (Rostammahabadi) [inline code/api/src/setup/schema/queries.js:23]: I remember an import of user_products here from the model, is that still necessary?

File: code/api/src/config/database.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+  },
+  "production": {
+    "username": "postgres",
+    "password": null,
+    "database": "crate",
+    "host": "127.0.0.1",
+    "dialect": "postgres",
+    "seederStorage": "sequelize"
+  }
+}
```
Comment (Rostammahabadi) [inline code/api/src/config/database.json:18]: Can we add this to the .gitignore file?

File: code/api/src/setup/schema/queries.js (hunk 7 above)
Comment (EricLarson2020) [reply code/api/src/setup/schema/queries.js:23]: The UserProducts has do queries so it shouldn't affect anything if it is deleted

File: code/api/src/config/database.json (hunk 8 above)
Comment (EricLarson2020) [reply code/api/src/config/database.json:18]: sure, will do

Comment (stellakunzang) [reply code/api/src/config/database.json:18]: As I wrote on Carlos' PR, trying to add to .gitignore after creating a PR is tricky! It took me about an hour to figure it out googling but I don't know what actually made the file disappear from the PR in the end. Sorry, wish I had more information!

File: code/api/src/migrations/7-user_products.js (hunk 1 above)
Comment (stellakunzang) [reply code/api/src/migrations/7-user_products.js:1]: This is awesome. So nice to see something that is readable!

File: code/api/src/modules/user/types.js (hunk 5 above)
Comment (stellakunzang) [reply code/api/src/modules/user/types.js:2]: Cool. I don't think I fully cognized that you need to import each individual type. Good to know!

File: code/api/src/migrations/7-user_products.js (hunk 1 above)
Comment (carflor) [reply code/api/src/migrations/7-user_products.js:1]: Thanks for getting this done!

File: code/api/src/modules/user/types.js (hunk 5 above)
Comment (carflor) [reply code/api/src/modules/user/types.js:2]: Looks good.

**Pull request merged**